packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2865,6 +2924,30 @@ export class ContainerRuntime
 			});
 		}
 
+		let idRange: IdCreationRange | undefined;
+		// Resubmited ops will already have an idRange specified and taking the next creation range
+		// again will result in finalizations out of order
+		if (this.runtimeOptions.enableRuntimeIdCompressor === true) {
+			idRange = this.idCompressor?.takeNextCreationRange();
+			// Don't include the idRange if there weren't any Ids allocated
+			idRange = idRange?.ids?.first !== undefined ? idRange : undefined;
+		}
+
+		let idAllocationBatchMessage: BatchMessage | undefined;
+		if (idRange !== undefined) {
+			const idAllocationMessage: ContainerRuntimeMessage = {
+				type: ContainerMessageType.IdAllocation,
+				contents: idRange,
+			};
+			idAllocationBatchMessage = {
```

Comment:
Why is the logic of adding state to the id allocation message split between here and the outbox? You could do them together in one place. Having this in 2 places increases the chance of introducing inconsistencies.

Reply:
I see. You don't want to send the stashed state over the wire.

Reply:
I'm not happy with the stashed op implementation right now. I have [AB#4042](https://dev.azure.com/fluidframework/internal/_workitems/edit/4042) for fixing it up.